### ISSUE TITLE
Bug: When uploading an image with a long link it throws an "Source is too long." error

### ISSUE THREAD

**Issue by neobooru** [rr-/szurubooru#272]
![image](https://user-images.githubusercontent.com/50623835/61955246-d98a2d00-afba-11e9-8fc1-84d27406b606.png)

This happens with (for example) DeviantArt links. 
We could just silently ignore those errors, or we could ask if the user wants to proceed without the source being set, thoughts?

**Comment by sgsunder** [comment]
I haven't examined this in detail yet so these are my first thoughts:

The `source` attribute for each post is stored as a size-200 varchar in the database, so I'm guessing this is the root cause of the error message. So the question becomes if we should leave this size or perform a database migration to increase it. This [StackOverflow](https://stackoverflow.com/q/417142) post suggests that there isn't a specific limitation for URL length but ~2000 is a practical maximum.

The first question we should ask is what should be the maximum URL length we want to support for the `source` attribute? I don't want to make it 2000 because that would create a large performance penalty just to address a non-critical edge case, but I'm not sure if we should leave it at 200 either.

Any thoughts?

**Comment by neobooru** [comment]
The DeviantArt urls I was talking about are usually (always?) below 800 chars, and I haven't seen any other sites that have such outrageously long urls. The easiest solution would be to up the source-length to 1k or 2k. This way the user never has an error or a missing source property. Performance wise it shouldn't make that much of a difference because a varchar doesn't use the storage space it doesn't need. We could even use varchar without specifying a length (but that is PostgreSQL specific, so that would break any compatibility with e.g. MySQL).

Types: char, varchar and text

> There are no performance differences between these three types, apart from increased storage size when using the blank-padded type, and a few extra cycles to check the length when storing into a length-constrained column. While character(n) has performance advantages in some other database systems, it has no such advantages in PostgreSQL. In most situations text or character varying should be used instead.

[Source](https://www.postgresql.org/docs/11/datatype-character.html)

tl;dr we should probably just set the source attribute length to 2000.

**Comment by sgsunder** [comment]
Sounds good. We'll migrate `source` to `VARCHAR(2048)` (might as well round to a power of 2) and if possible I also want to migrate `post_flags` down to `VARCHAR(32)` because we never use more than 10 characters for this record.

Do you want me to do this or do you want to make the PR yourself?

**Comment by neobooru** [comment]
You probably know where to look, so I'd say let you do it. I personally wouldn't touch the post_flags length, because why bother with that? And if we ever have new tags we'd have to scale it back up.
